### ISSUE TITLE
Simulation Experiment view: the runtime doesn't get updated after having edited a CellML file

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#1385]
Following our 2017-07-23 snapshot, we can run a model using the `Simulation Experiment` view. However, if we switch to, say, the `CellML Text` view and modify one of the model constants, then when switching back to the `Simulation Experiment` view, we would expect the list of model parameters to be updated accordingly and it isn't. (Thanks to @ssafaei for the report.)

FWIW, this is most likely related to our work on issue #1266.

**Comment by dbrnz** [comment]
FWIW, @span17 has also noticed this issue...

**Comment by agarny** [comment]
Yes, my bad. I tested that case, but then addressed another issue, which broke that case.I have since fixed the problem, but I am not happy with my solution, so I am going to implement a better one, which will result in our `fileReloaded()` method from our file handling interface to be reverted. Indeed, to get issue #1266 to work, I changed that method from

```
virtual void fileReloaded(const QString &pFileName,
                          const bool &pFileChanged) PURE;
```

to

```
virtual void fileReloaded(const QString &pFileName,
                          const bool &pFileChanged,
                          const bool &pFileJustSaved) PURE;
```

but I really don't like it. It just seems odd to me that one would have to specify whether a file has just been saved.

Anyway, I believe I have a solution. I just need to implement it once I am back home.
